File: java/socketd/src/main/java/org/noear/socketd/transport/core/entity/StringEntity.java
package org.noear.socketd.transport.core.entity;

import java.nio.charset.StandardCharsets;

/**
 * 字符串实体
 *
 * @author noear
 * @since 2.0
 */
public class StringEntity extends EntityDefault {
    public StringEntity(String str) {
        dataSet(str.getBytes(StandardCharsets.UTF_8));
    }
}

File: javascript/socketd/src/socketd/transport/core/entity/StringEntity.ts
import {EntityDefault} from "./EntityDefault";
import {Entity} from "../Entity";
import {StrUtils} from "../../../utils/StrUtils";

/**
 * 字符串实体
 *
 * @author noear
 * @since 2.0
 */
export class StringEntity extends EntityDefault implements Entity {
    constructor(data: string) {
        super();
        const dataBuf = StrUtils.strToBuf(data);
        this.dataSet(dataBuf);
    }
}